host_tools/fw_update.py

```diff
@@ -80,18 +102,26 @@ def scan_for_device(self):
         ports = [x[0] for x in list_ports.comports()]
         retport = None
         self._log("Searching for an active bootloader")
-        
+
         for p in ports:
             try:
                 rsp = self._send_ping(p)
                 if not rsp == None:
-                    self._log("Found bootloader on: {}".format(p))
+                    ver = self._get_version(p)
+                    if not ver:
+                        version = '<=0.1.2'
+                    else:
+                        version = '.'.join(
```

Comment:
This object is autogenerated.  It has a str object, but it is a lot more verbose than just printing the concatenated 'x.x.x' format.  Since the autogen system has no way to know what exactly a person wants to do with the end result objects, it's not really possible to generically have this sort of method.